client/src/components/ContentNavigator/index.ts

```diff
@@ -247,6 +250,71 @@ class ContentNavigator implements SubscriptionProvider {
           "workbench.actions.treeView.contentdataprovider.collapseAll",
         );
       }),
+      commands.registerCommand(
+        "SAS.convertSasnbToFlw",
+        async (resource: ContentItem) => {
+          const resourceUri = getUri(resource);
+          // Ensure that the command is only available for .sasnb files
+          if (resourceUri.path.endsWith(".sasnb")) {
+            // Open window to chose the name and location of the new .flw file
+            const name = await window.showInputBox({
+              prompt: "Enter the name for the new .flw file",
+              value: resource.name.replace(".sasnb", ".flw"),
+            });
+
+            if (!name) {
+              // User canceled the input box
+              return;
+            }
+
+            if (!name.endsWith(".flw")) {
+              window.showErrorMessage(Messages.InvalidFlwFileNameError);
+              return;
+            }
+
+            // check that name does not exist
+            const parent = await this.contentDataProvider.getParent(resource);
+            const children = await this.contentDataProvider.getChildren(parent);
+            if (children.some((child) => child.name === name)) {
+              window.showErrorMessage(Messages.FileAlreadyExistsError);
+              return;
+            }
+
+            try {
+              const contentUri: string =
+                await this.contentDataProvider.provideTextDocumentContent(
+                  resourceUri,
+                );
+              const flowDataUint8Array = convert_sasnb_to_flw(contentUri, name);
+              // error if content is empty
+              if (flowDataUint8Array.length === 0) {
+                window.showErrorMessage(Messages.NoCodeToConvert);
+                return;
+              }
+              const newUri = await this.contentDataProvider.createFile(
+                parent,
+                name,
+              );
+              await this.contentDataProvider.writeFile(
+                newUri,
+                flowDataUint8Array,
+              );
+              await this.contentDataProvider.associateFlow(
+                name,
+                newUri,
+                parent,
+              );
+              window.showInformationMessage(
+                Messages.SasnbToFlwConversionSuccess,
+              );
+            } catch (error) {
+              window.showErrorMessage(Messages.SasnbToFlwConversionError);
+            }
+          } else {
+            window.showErrorMessage(Messages.InvalidSasnbFileError);
+          }
```

Comment:
Can we make this a method inside `ContentDataProvider`?

Reply:
done in 2c900b3fdb3790fe93dce7067ca4ebb5ed93d544. I have also moved the `handleCreationResponse` function to `ContentDataProvider`